nasa-jpl/explorer-1
# Feature: adding Internal Header and Internal Footer

### Checklist

  - PR titles are used to generate the change log in [releases](../releases); good ones make that easier to scan.
  - PRs will be broadly categorized in the change log, but for even easier scanning, consider prefixing with a component name or other useful categorization, e.g., "BaseButton: fix layout bug", or "Storybook: Update dependencies".
  - These tags are used to do the aforementioned broad categorization in the change log and determine what the next release's version number should be.
  - [Release Drafter](https://github.com/marketplace/actions/release-drafter) will attempt to do the category labeling for you! Please double-check its work.

## Description

Resolves #227. This PR grew pretty massive as adding the internal header and footer required:
- new components
- modifying/updating old components
- new component organization
- new scss
- new javascript

It's a lot to review, and I wouldn't be surprised if I missed something. Overall, I think the organization makes sense and lets users create their own header, but I think switching between canvas view and docs view doesn't make for the best experience when trying to compare configurations. Hopefully it's clear enough for now and also sets a foundation for how to we want to structure the external header and footer. Really open to any input on the documentation, but I was largely focused on getting this to a "good enough to start with" state so we could move on to other tasks.

I wasn't sure what to call the parent category. I settled on "Global Layout." I also read that "Marginals" is technically correct, but I don't think it's very clear. I felt they needed to be separate from the overall "Components" category though. The story levels also nest pretty deep in this section as well, so it would be nice to not nest them further.

I'm out of time today, so I'll work on writing a detailed write-up of what changed on the next work day.

## Instructions to test

1. `npm run storybook`
2. Check the stories in the [Global Layout](http://localhost:6006/?path=/docs/global-layout-overview--page) section. Note that Javascript functionality won't work correctly unless you are in canvas view.
3. Check stories for other components that were also modified (BaseLink, for example). It would be also good to check stories that use modified components within them.

## Migration Notes

Really only for WCP: class names for the menu templates have changed. Update the class names to match (more details to come that I'll write up on the next work day).

By default, `.container` now has a default padding of `1rem`. For the most part, this should not affect your projects, unless you have built in your own paddings, or used any of the named grid lines for MixedBleed layouts. If so, you can easily override this globally in your tailwind config, or, to address specific use-cases, add `.px-0` to the `.container` in question.

To override the container padding globally, add this to your Tailwind config:

```js
// your-project/tailwind.config.js

// import Explorer 1's Tailwind config
const explorer1Config = require('@nasa-jpl/explorer-1/tailwind.config.js')

module.exports = {
  ...explorer1Config,
  theme: {
    ...explorer1Config.theme,
    container: {
      ...explorer1Config.theme.container
      padding: '0', // ⬅️ override padding here
    },
  purge: ['../**/*.html'], // your project's purge settings
}
```

## Tested in the following environments/browsers:

### Operating System


### Browser
